lib/AppInfo/Application.php

```diff
@@ -74,4 +75,17 @@ protected function registerUserInterface(): void {
 			Util::addStyle('notifications', 'styles');
 		}
 	}
+
+	protected function registerUserDeleteHook(): void {
+		Util::connectHook('OC_User', 'post_deleteUser', $this, 'deleteUser');
+	}
+
+	public function deleteUser(array $params): void {
+		$manager = $this->getContainer()
+			->getServer()
+			->getNotificationManager();
+		$notification = $manager->createNotification();
+		$notification->setUser($params['uid']);
+		$manager->markProcessed($notification);
```

Comment:
Mind elaborating/commenting why this works? Does this rely on some other logic that cleans up other notifications if you do not set other properties?

Reply:
yeah basically the notification given is a filter. It will "mark processed" all matching notifications. if they dont have an id or any other data, but only a user. everything for the user will be deleted.

But now that you say it, we can instead use the dedicated method of the handler already.

Reply:
Nice. thanks!